### DISCUSSION TITLE
Problem reproducing secreted metabolites in solution.

### DISCUSSION THREAD

**Discussion by lkroeh** [opencobra/cobrapy#1108] [Q&A]
Hi, 

I'm using the human RECON2 model, where I load in the model, change the objective function to biomass, change the media, and then do `solution = model.optimize()`; followed by `print(model.summary(solution=solution))`.  I see lactate is secreted in the summary.  
When I do `solution.fluxes.EX_lac_L_b` (the reaction for lactate production) I can confirm there is a flux.
However, when i do `model.metabolites.get_by_id('lac_L_b').summary()`, there is no information reported!

Additionally, when I re-run `print(model.summary(solution=solution))`, I not longer see lactate as a secreted product, although the biomass solution value does not change. 

So 2 questions:
1.  Why when I see lactate secretion reported in the summary, can I can not find information on what reaction is producing it?
2.  Why are the secreted products changing when I re-run the code chunk which defines the media (but without changing any values from the run before) and print the solution?

Thanks for your help!!
Lina

**Comment by Midnighter** [comment]
Hi Lina,

Without having looked at it, my guess is that this is due to there being many different flux distributions that enable the same optimal objective value (non-unique solution). You can confirm this by also using FVA in the summaries. You should see that there many potential candidates for being secreted, not only lactate.

You can get a more reproducible solution by using parsimonious FBA, although that still does not necessarily give you a unique solution. Do you just want to understand the model better or do you have a particular biological question that you would like to answer/model?

**Reply by lkroeh** [reply]
Hi, Thank you Midnighter!! You're very right, I will use FVA/pFBA.  I just want to understand the model better, but am also now intrigued by the lactate production that I sometimes see, so also want to investigate why that may be happening.
Best, 
Lina
